files/initramfs-tools/union-mount.j2

```diff
@@ -161,3 +161,11 @@ fi
 if [ -f /tmp/fsck.log.gz ]; then
     mv /tmp/fsck.log.gz ${rootmnt}/var/log
 fi
+
+## Move sonic-environment to /etc/sonic
+sonic_config_dir=${rootmnt}/host/${image_dir}/sonic-config
+sonic_env_file=${sonic_config_dir}/sonic-environment
+if [ -d ${sonic_config_dir} -a -f ${sonic_env_file} ]; then
+    echo "moving file ${sonic_env_file} to ${rootmnt}/etc/sonic" 1>&2
+    mv ${sonic_env_file} ${rootmnt}/etc/sonic
+fi
```

Comment:
I am not exactly sure why we do this in the union-mount? i thought the first time we need those variable is after rc.local? then, can we do this in rc.local? it just easier to debug.

Reply:
Yes. Both rc-local service and ntp service are the first service to call into sonic-cfggen. I think last I checked they both start after networking service. The reason to put here is that I want to address both rc.local and ntp use. I have disabled ntp service systemd start and not there is no contention. I will move it to rc.local.